lib/Dialect/FIRRTL/Transforms/ExtractInstances.cpp

```diff
@@ -159,12 +163,18 @@ struct ExtractInstancesPass
   CircuitNamespace circuitNamespace;
   /// Cached module namespaces.
   DenseMap<Operation *, hw::InnerSymbolNamespace> moduleNamespaces;
+  /// The metadata class ops.
+  ClassOp extractMetadataClass, schemaClass;
+  const unsigned prefixNameFieldId = 0, pathFieldId = 2, fileNameFieldId = 4;
+  /// Cache of the inner ref to the new instances created. Will be used to
+  /// create a path to the instance
+  DenseMap<InnerRefAttr, InstanceOp> innerRefToInstances;
```

Comment:
this, and the `extractMetadataClass` and `schemaClass` variables should be reset in `runOnOperation` for good hygiene, even if we only ever run this pass once on the whole IR.